woocommerce/woocommerce-gutenberg-products-block
# Add Mini Cart block in experimental builds

Part of #4138 and #4245.

This PR adds to the experimental builds an initial version of a MiniCart block which has the following characteristics:

* Only a small JavaScript file is downloaded & parsed on each page where the Mini Cart block is present. That script is used to detect mouse hover/blur on the mini cart button and to preload + append the other scripts.
* Once the page is loaded, all other scripts are preloaded. That means they don’t block rendering the page and, instead, they are downloaded in the background after the page finishes loading.
* Once the user interacts with the Mini Cart, those scripts are appended to the page and, because of that, parsed and executed.

The benefits of this approach are:

* We don't block page rendering to download the scripts required to render the block: scripts are downloaded in the background.
* Files are only parsed and executed when necessary.
* Using the same components and hooks as in the Cart block, we can reuse all the logic. Extensibility points can be reused as well.

This approach was discussed in pca54o-1PL-p2 (conclusions can be found in https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4245#issuecomment-899394073).

## How to test the changes in this Pull Request:

1. Create a post or page and add a Mini Cart block.
2. Visit that post or page in the frontend and verify the Mini Cart button appears and, when hovered, Cart contents are rendered.

This PR is an initial step, so don't expect the block to look or work perfectly. UI, interactions, etc. will be improved in subsequent PRs. This epic has some of the next steps we will be working on: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4138.
